Documentation for new Kalpana app (#209)
DesignSafe-CI/DS-User-Guide: user-guide/docs/tools/visualization/kalpana.md

## Kalpana
<a name="kaplana-user-guide"></a><!-- old heading name/id -->

> Kalpana is a Python library which primarily converts ADCIRC NetCDF output files to GIS-compatible shapefiles. Secondary functions are not available in the DesignSafe application. It was developed by the Coastal and Computational Hydraulics Team at North Carolina State University. More information regarding Kalpana can be found at:

Reviewer: 1. Consider removing "regarding Kalpana", because user is reading doc for Kaplana, so the context seems obvious.
2. Consider removing "can be found at", because it does not seem to assist the reader. The phrase ends with a colon, so a complete sentence is not necessary.

```suggestion
Kalpana is a Python library which primarily converts ADCIRC NetCDF output files to GIS-compatible shapefiles. Secondary functions are not available in the DesignSafe application. It was developed by the Coastal and Computational Hydraulics Team at North Carolina State University. More information:
```

<ul>
	<li>The CCHT website: <a href="https://ccht.ccee.ncsu.edu/kalpana/" target="_blank">https://ccht.ccee.ncsu.edu/kalpana/</a></li>
	<li>The official GitHub repository, which contains the latest release and examples: <a href="https://github.com/ccht-ncsu/Kalpana" target="_blank">https://github.com/ccht-ncsu/Kalpana</a></li>

> </ul>

Reviewer: 1. New content uses Markdown syntax.
2. We can just link name of page to the URL.

```suggestion

- [Kaplana page on CCH website](https://ccht.ccee.ncsu.edu/kalpana/)
- [official GitHub repository](https://github.com/ccht-ncsu/Kalpana), which contains the latest release and examples
```

### Submitting a Job

#### On the Web Portal

> Generating contours for a single case is simple through the web interface. The following information is required:

Reviewer: 1. <details><summary>To streamline content digestion for user…</summary>

    - reduces verbosity
    - avoids "Required" being ignored (by a reader skimming) because it:
        - is within a sentence reader may skim over
        - may appear on the right side of the page[^1]

   </details>

[^1]: Readers that skim may be scanning left side of page for headings, sentence starts, list introductions, et cetera.

```suggestion
Generating contours for a single case is simple through the web interface.

Required Information:
```

<ul>
	<li><b>ADCIRC File.</b> See the reference table below. Generic NetCDF files may also be accepted, if formatted appropriately.</li>
	<li><b>Variable.</b> Preceded by flag --var. The variable of interest from the NetCDF file. See the reference table below.</li>
	<li><b>Contour levels.</b> Preceded by flag --levels. Separated by spaces, the minimum contour, maximum contour, and step size.</li>
	<li><b>Vertical units out.</b> m or ft</li>
	<li><b>Vertical units in.</b> m or ft</li>
	<li><b>Time Steps.</b> Preceded by flag --timesteps. For time-varying files, specific time steps to extract are provided, separated by spaces. Otherwise, all are exported.</li>

> </ul>

Reviewer: 1. New content uses Markdown syntax.
2. Using back-ticks around API/code terms distinguish them as such.

```suggestion

- **ADCIRC File.** See the reference table below. Generic NetCDF files may also be accepted, if formatted appropriately.
- **Variable.** Preceded by flag `--var`. The variable of interest from the NetCDF file. See the reference table below.
- **Contour levels.** Preceded by flag `--levels`. Separated by spaces, the minimum contour, maximum contour, and step size.
- **Vertical units out.** m or ft
- **Vertical units in.** m or ft
- **Time Steps.** Preceded by flag `--timesteps`. For time-varying files, specific time steps to extract are provided, separated by spaces. Otherwise, all are exported.
```

After this, generic allocation and archiving preferences are available. Since Kalpana is a serial program, it does not make sense to run on multiple nodes. However, additional cores may provide additional memory to process very large files.

<table>

> 	<caption>Reference Table for Common ADCIRC Files</caption>

Reviewer: I avoid converting table to Markdown yet, because `<caption>` would be lost.[^1]

[^1]: I'll convert after [`mkdocs-caption`](https://github.com/tobiasah/mkdocs-caption) is installed.

			<td>depth, u-vel, v-vel</td>
		</tr>
		<tr>
			<td>fort.73.nc</td>
			<td>Time-dependent air pressure</td>
			<td>depth, pressure</td>
		</tr>
		<tr>
			<td>fort.74.nc</td>
			<td>Time-dependent wind velocity</td>
			<td>depth, windx, windy</td>
		</tr>
		<tr>
			<td>maxele.63.nc</td>
			<td>Maximum water elevation</td>
			<td>depth, time_of_zeta_max, zeta_max</td>
		</tr>
		<tr>
			<td>maxvel.63.nc</td>
			<td>Maximum water velocity</td>
			<td>depth, time_of_vel_max, vel_max</td>
		</tr>
		<tr>
			<td>maxwvel.63.nc</td>
			<td>Maximum wind velocity</td>
			<td>depth, time_of_wind_max, wind_max</td>
		</tr>
		<tr>
			<td>swan_HS.63.nc</td>
			<td>Significant wave height</td>
			<td>swan_HS</td>
		</tr>
		<tr>
			<td>swan_HS_max.63.nc</td>
			<td>Maximum significant wave height</td>
			<td>swan_HS_max</td>
		</tr>
		<tr>
			<td>swan_TPS.63.nc</td>
			<td>Peak wave period</td>
			<td>swan_TPS</td>
		</tr>
		<tr>
			<td>swan_TPS_max.63.nc</td>
			<td>Maximum peak wave period</td>
			<td>swan_TPS_max</td>
		</tr>
		<tr>
			<td>swan_TMM10.63.nc</td>
			<td>Mean wave period</td>
			<td>swan_TMM10</td>
		</tr>
	</tbody>
</table>

#### Advanced Functionality with Tapis

> To generate shapefiles for an ensemble, or to modify advanced Kalpana settings, submit JSON job requests using TapiPy (<a href="https://github.com/tapis-project/tapipy" target="_blank">https://github.com/tapis-project/tapipy</a>). Descriptions for each flag are available in the Tapis app specification or the source file on GitHub, see function nc2shp (<a href="https://github.com/ccht-ncsu/Kalpana/blob/master/kalpana/export.py" target="_blank">https://github.com/ccht-ncsu/Kalpana/blob/master/kalpana/export.py</a>).

Reviewer: 1. New content uses Markdown syntax.
2. We can just link name of content to the URL.
3. Consider linking directly to the function, so user need not search.
4. Consider linking to current version of code, so future refactors do not make user guide inaccurate.

```suggestion
To generate shapefiles for an ensemble, or to modify advanced Kalpana settings, submit JSON job requests using [TapiPy](https://github.com/tapis-project/tapipy){ target="_blank" }. Descriptions for each flag are available in the Tapis app specification or the source file on GitHub, see [function `nc2shp`](https://github.com/ccht-ncsu/Kalpana/blob/v0.0.25/kalpana/export.py#L764){ target="_blank" }).

```
